.github/workflows/python-client.yml

```
@@ -0,0 +1,84 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+# This workflow will build a Python project with Poetry and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://docs.github.com/en/actions/use-cases-and-examples/building-and-testing/building-and-testing-python
+
+name: Python CI with Poetry
+
+on:
+  push:
+    branches: [ "main" ]
+    paths:
+      - 'client/python/**'
```

```suggestion
  push:
    branches: [ "main" ]
```
I think it will be good to always trigger the CI when there is a new commit in main. This allows us to check the correctness periodically to ensure there is no changes in other folder that unexpectedly break the CLI.

Personally, I am in favor of removing the `paths` parameter in pull_request too to make the CI config aligned with other workflow:
```
on:
  push:
    branches: [ "main" ]
  pull_request:
    branches: [ "main" ]
```